File: qt/gen_qlockfile.cpp
// +build ignore

#include <QLockFile>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qlockfile.h>
#include "gen_qlockfile.h"

QLockFile* QLockFile_new(struct miqt_string fileName) {
	QString fileName_QString = QString::fromUtf8(fileName.data, fileName.len);
	return new QLockFile(fileName_QString);
}

struct miqt_string QLockFile_FileName(const QLockFile* self) {
	QString _ret = self->fileName();
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

bool QLockFile_Lock(QLockFile* self) {
	return self->lock();
}

bool QLockFile_TryLock(QLockFile* self, int timeout) {
	return self->tryLock(static_cast<int>(timeout));
}

void QLockFile_Unlock(QLockFile* self) {
	self->unlock();
}

void QLockFile_SetStaleLockTime(QLockFile* self, int staleLockTime) {
	self->setStaleLockTime(static_cast<int>(staleLockTime));
}

int QLockFile_StaleLockTime(const QLockFile* self) {
	return self->staleLockTime();
}

bool QLockFile_TryLock2(QLockFile* self) {
	return self->tryLock();
}

bool QLockFile_IsLocked(const QLockFile* self) {
	return self->isLocked();
}

bool QLockFile_RemoveStaleLockFile(QLockFile* self) {
	return self->removeStaleLockFile();
}

LockError QLockFile_Error(const QLockFile* self) {
	return self->error();
}

void QLockFile_Delete(QLockFile* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QLockFile*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qlockfile.go
package qt

import (
	"unsafe"
)

type QLockFile__LockError int

const (
	QLockFile__NoError         QLockFile__LockError = 0
	QLockFile__LockFailedError QLockFile__LockError = 1
	QLockFile__PermissionError QLockFile__LockError = 2
	QLockFile__UnknownError    QLockFile__LockError = 3
)

type QLockFile struct {
	h          uintptr
	isSubclass bool
}

// NewQLockFile constructs a new QLockFile object.
func NewQLockFile(fileName string) *QLockFile {
	fileName_ms := struct_miqt_string{}
	fileName_ms.data = CString(fileName)
	fileName_ms.len = size_t(len(fileName))
	defer free(unsafe.Pointer(fileName_ms.data))
	g := newQLockFile(QLockFile_new(fileName_ms))
	g.isSubclass = true
	return g
}

func (this *QLockFile) FileName() string {
	var _ms struct_miqt_string = QLockFile_FileName(this.h)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QLockFile) Lock() bool {
	return (bool)(QLockFile_Lock(this.h))
}

func (this *QLockFile) TryLock(timeout int) bool {
	return (bool)(QLockFile_TryLock(this.h, (int)(timeout)))
}

func (this *QLockFile) Unlock() {
	QLockFile_Unlock(this.h)
}

func (this *QLockFile) SetStaleLockTime(staleLockTime int) {
	QLockFile_SetStaleLockTime(this.h, (int)(staleLockTime))
}

func (this *QLockFile) StaleLockTime() int {
	return (int)(QLockFile_StaleLockTime(this.h))
}

func (this *QLockFile) TryLock2() bool {
	return (bool)(QLockFile_TryLock2(this.h))
}

func (this *QLockFile) IsLocked() bool {
	return (bool)(QLockFile_IsLocked(this.h))
}

func (this *QLockFile) RemoveStaleLockFile() bool {
	return (bool)(QLockFile_RemoveStaleLockFile(this.h))
}

func (this *QLockFile) Error() LockError {
	xxxxxxxxx
}
